R/make_module_macro.R

```diff
@@ -0,0 +1,53 @@
+#' Generate extendr module macro for Rust source
+#'
+#' Read some Rust source code, find functions or implementations with the
+#' `#[extendr]` attribute, and generate an `extendr_module!` macro statement.
+#'
+#' This function uses simple regular expressions to do the Rust parsing and
+#' can get confused by valid Rust code. It is only meant as a convenience for
+#' simple use cases. In particular, it cannot currently handle implementations
+#' for generics.
+#' @param code Character vector containing Rust code.
+#' @param module_name Module name
+#' @keywords internal
+#' @export
+make_module_macro <- function(code, module_name = "rextendr") {
+  # make sure we have cleanly separated lines
+  lines <- stringi::stri_split_lines(glue::glue_collapse(code, sep = "\n"), omit_empty = TRUE)
+  idents <- get_extendr_idents(lines[[1]])
+
+  outlines <- c("extendr_module! {", glue::glue("mod {module_name};"))
+  outlines <- c(outlines, glue::glue_data(idents, "{type} {ident};"))
+  outlines <- c(outlines, "}")
+  outlines
+}
+
+
+get_extendr_idents <- function(lines) {
+  # find all lines with the "#[extendr]" decoration
+  idx <- which(grepl("\\#\\[extendr\\]", lines))
+  idx <- idx + 1 # bump indices by one to look in subsequent lines
```

Comment:
Is it possible to have another attribute between `#[rextendr]` and function declaration?
I see that empty lines are dropped, but what about commented-out lines?
It does not make sense for Rust one-liners, but if you store some rust functions in a text file and send this text as an input, it may very well contain poorly placed comments.
```Rust
#[rextendr]
// Not good but still valid, right?
fn unmatched_fun() {}
```
If you feel like this may be important, perhaps we should identify commented-out lines as-well.

Reply:
Ok, added code to strip all lines that are fully commented out. I could also add code to strip all lines that contain an attribute that is not `#[extendr]`, if you can think of a clever regex that identifies those lines. Otherwise I'll leave it as a TODO for the future.

Reply:
@clauswilke,
I suggest leave it as todo to speed up things, as extra attributes are much rarer than just comments. 
We should also probably error if we are unable to find exports (i.e. malformed rust source), but this can be verified later.

I will play around with some regex and report to your todo issue.